src/modules/ranking-data.js

```diff
@@ -0,0 +1,82 @@
+import sequelize from 'sequelize';
+import redisClient from '../config/redis.js';
+import BossRaidRecord from '../models/bossraid-record.js';
+import BossRaid from '../models/bossraid.js';
+import { rainkingDataKey } from '../utils/constants.js';
+
+const getRankingData = async (userId) => {
+  const topRankerInfo = await getTopRankerInfo();
+  const myRankingInfo = getMyRankingInfo(topRankerInfo, userId);
+
+  return {
+    topRankerInfo,
+    myRankingInfo,
+  };
+};
+
+/*
+  Top ranking data 가져오기
+  1. 캐시 여부 확인
+  2.1. 캐시가 있다면 레디스에 캐싱된 정보 반환
+  2.2. 캐시가 없다면 랭킹 구한 후 캐싱 및 반환
+*/
+const getTopRankerInfo = async (userId) => {
+  let topRankerInfo;
+
+  const isCached = await redisClient.exists(rainkingDataKey);
+  if (isCached) {
+    console.log('캐싱된 데이터 가져오기');
+    topRankerInfo = await redisClient.get(rainkingDataKey);
+  } else {
+    console.log('랭킹 구하기 및 캐싱 후 데이터 가져오기');
+    topRankerInfo = await cachingTopRankerInfo(userId);
+  }
+
+  return JSON.parse(topRankerInfo);
+};
+
+const cachingTopRankerInfo = async (userId) => {
+  let topRankerInfo = await BossRaidRecord.findAll({
+    raw: true,
+    attributes: [
+      [
+        sequelize.cast(sequelize.fn('sum', sequelize.col('score')), 'float'),
+        'totalScore',
+      ],
+    ],
+    include: {
+      model: BossRaid,
+      attributes: ['user_id'],
+    },
+    group: ['user_id'],
+    limit: 5,
+  });
+
+  topRankerInfo.forEach((item, index) => {
+    item.userId = item['BOSSRAID.user_id'];
+    delete item['BOSSRAID.user_id'];
+
+    item.isUserId = item.userId === parseInt(userId) ? true : false;
```

Comment:
`item.isUserId = item.userId === parseInt(userId) ? true : false;`
`item.isUserId = item.userId === +userId ? true : false;`
이렇게 바꾸시는건 어떠신가요??

Reply:
@KCH6937 
넵! 다시 보니까 굉장히 헷갈리게 작성한 것 같네요...!
```suggestion
    item.isUserId = (item.userId === parseInt(userId)) ? true : false;
```
이렇게 바꾸는건 어떨까요???

Reply:
그것도 좋은 것 같습니다!!
`item.isUserId = item.userId === +userId` 이렇게 단축이 가능할 듯 해서 말씀드렸습니다.
다시보니 예진님 코드가 가독성이 훨씬 좋아보이네요!